src/scraper.ts

```diff
@@ -0,0 +1,57 @@
+import { load, type CheerioAPI } from "cheerio";
+import decodeEntities from "html-entities-decode";
+
+export type CheerioNode = {
+	name: string;
+	children: CheerioNode[];
+	attribs: { [attr: string]: string };
+};
+
+if (process.env.SCRAPE_CACHE) {
+	if (!Number.isInteger(+process.env.SCRAPE_CACHE))
+		throw new TypeError(
+			"Environment variable SCRAPE_CACHE should be an integer",
+		);
+	if (+process.env.SCRAPE_CACHE < 1)
+		throw new RangeError(
+			"Environment variable SCRAPE_CACHE should be a positive integer",
+		);
+}
+
+const CACHE_DURATION = +(process.env.SCRAPE_CACHE || 1) * 3600_000;
+const cache = new Map<string, CheerioAPI>();
+
+export default async function scrape(url: string, skipCache = false) {
+	if (!skipCache) {
+		const cached = cache.get(url);
+		if (cached) return cached;
+	}
+
+	const html = await fetch(url).then((r) => r.text());
+	const crawler = load(html);
+	cache.set(url, crawler);
+	setTimeout(cache.delete.bind(cache, url), CACHE_DURATION);
```

Comment:
This is not a great practice, as it will not handle restarts. What I have done previously for something similar is to create a request from the URL (do not use the existing request) and set an `X-Date` header to either the creation date or expiration date, and then whenever that URL is loaded from cache, check the X-Date header. This will also allow us to implement SWR if we want, although I am not sure if that is a great idea.

Reply:
I'm not sure I understand. You want to prepare a request to the URL and store that in the cache?
Wouldn't it be easier to just store the expiration date and skip the cache if we're past it? Something like
```ts
const now = Date.now();
const cached = cache.get(url);
if(cached?.expiration < now) return cached.crawler;
const html = await fetch(url).then((r) => r.text());
const crawler = load(html);
cache.set(url, { expiration: now + CACHE_DURATION, crawler });
return cache;```

Reply:
That looks like what I suggested, but weird. I'll go with what I wrote, but with storing the fetching date instead of expiration date as you suggested.